chainer/functions/normalization/batch_normalization.py

```diff
@@ -56,27 +56,38 @@ def check_type_forward(self, in_types):
                 '%s == %s' % (in_types.size(), n_in))
         x_type, gamma_type, beta_type = in_types[:3]
         M = gamma_type.ndim.eval()
-        type_check.expect(
-            x_type.dtype.kind == 'f',
-            x_type.ndim >= gamma_type.ndim + 1,
-            x_type.shape[1:1 + M] == gamma_type.shape,
-            # TODO(beam2d): Check shape
-            gamma_type.dtype == x_type.dtype,
-            beta_type.dtype == x_type.dtype,
-            gamma_type.shape == beta_type.shape,
-        )
+        if x_type.dtype.eval() == numpy.float16:
```

Comment:
I will fix the code based on your suggestion.
BTW, I think that whether x is `numpy` or `cupy` is not easily determined in the method `check_type_forward`, because the argument is `in_types` and not `inputs`...  Any idea?